pymeasure/instruments/rigol/rigol_dg800.py

```diff
@@ -0,0 +1,167 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, Channel, SCPIMixin
+from pymeasure.instruments.validators import truncated_discrete_set
+
+
+class VoltageChannel(Channel):
+    """A channel of the signal generator"""
+
+    output_enabled = Channel.control(
+        ":OUTP{ch}?",
+        ":OUTP{ch} %s",
+        """Control the status of the channel.
+         True if the channel is on and False if not.""",
+        validator=truncated_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+    load = Channel.control(
+        ":OUTP{ch}:LOAD?",
+        ":OUTP{ch}:LOAD %f",
+        """Control the output impedance of a given channel.
+         Returns 9.9e+37 for infinite impedance.""",
+    )
+
+    high_impedance = Channel.control(
+        ":OUTP{ch}:IMP?",
+        ":OUTP{ch}:IMP %s",
+        """Control the output impedance to be HighZ.
+        This means displayed voltages will be internally converted to
+        correspond that seen by an infinite load, not 50 Ohms.
+        See p. 82 of the manual for more information.""",
+        validator=truncated_discrete_set,
+        values={True: "INF", False: "50"},
+        map_values=True,
+    )
+
+    frequency = Channel.control(
+        ":SOUR{ch}:FREQ?", ":SOUR{ch}:FREQ %f", """ Control the output frequency (Hz)"""
+    )
+
+    sync = Channel.control(
+        ":OUTP{ch}:SYNC?",
+        ":OUTP{ch}:SYNC %s",
+        """Control the synchronization flag.
+          True if the channel triggers the synchronization connector and False if not """,
+        validator=truncated_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    sine = Channel.setting(
+        ":SOUR{ch}:APPL:SIN %f,%f,%f,%f",
+        """
+        Set the waveform generator to output a sine of specified parameters
+        :param freq: (int) The frequency of the sine in Hz
+        :param ampl: (float) The peak to peak amplitude of the sine in V
+        :param offset: (float) The DC offset of the sine in V
+        :param phase: (float) The phase offset of the sine in degrees
```

Comment:
For the rendering, you can add the type inside the parameter:
```suggestion
        :param int freq: The frequency of the sine in Hz
        :param float ampl: The peak to peak amplitude of the sine in V
        :param float offset: The DC offset of the sine in V
        :param float phase: The phase offset of the sine in degrees
```
Please add that you have to give the parameters as a tuple. Something like `Set ... with a tuple of :param...`